impeller/entity/geometry/point_field_geometry.cc

```diff
@@ -79,44 +80,51 @@ PointFieldGeometry::GetPositionBufferCPU(const ContentContext& renderer,
   Scalar min_size = 1.0f / sqrt(std::abs(determinant));
   Scalar radius = std::max(radius_, min_size);
 
-  auto vertices_per_geom = ComputeCircleDivisions(
-      entity.GetTransform().GetMaxBasisLength() * radius, round_);
-  auto points_per_circle = 3 + (vertices_per_geom - 3) * 3;
-  auto total = points_per_circle * points_.size();
-  auto radian_start = round_ ? 0.0f : 0.785398f;
-  auto radian_step = k2Pi / vertices_per_geom;
-
   VertexBufferBuilder<SolidFillVertexShader::PerVertexData> vtx_builder;
-  vtx_builder.Reserve(total);
-
-  /// Precompute all relative points and angles for a fixed geometry size.
-  auto elapsed_angle = radian_start;
-  std::vector<Point> angle_table(vertices_per_geom);
-  for (auto i = 0u; i < vertices_per_geom; i++) {
-    angle_table[i] = Point(cos(elapsed_angle), sin(elapsed_angle)) * radius;
-    elapsed_angle += radian_step;
-  }
 
-  for (auto i = 0u; i < points_.size(); i++) {
-    auto center = points_[i];
-
-    auto origin = center + angle_table[0];
-    vtx_builder.AppendVertex({origin});
-
-    auto pt1 = center + angle_table[1];
-    vtx_builder.AppendVertex({pt1});
-
-    auto pt2 = center + angle_table[2];
-    vtx_builder.AppendVertex({pt2});
-
-    for (auto j = 0u; j < vertices_per_geom - 3; j++) {
-      vtx_builder.AppendVertex({origin});
-      vtx_builder.AppendVertex({pt2});
-
-      pt2 = center + angle_table[j + 3];
-      vtx_builder.AppendVertex({pt2});
+  if (round_) {
```

Comment:
This code needs to use vtx builder due to the unfortunate way we interleave vertex data. No action needed.

Reply:
I'm missing the distinction here. The vtx builder collects them interleaved in a vector and then inserts them "as is" into the host buffer. So, can't I use the same technique as the position data, but just insert adjacent points?

https://github.com/flutter/engine/blob/32a6383eace44cfa4890427d4e3576f3e6362ae2/impeller/renderer/vertex_buffer_builder.h#L108

Reply:
I ended up using HostBuffer injection. We are asking it to allocate space for N x TexturePerVertexData objects so that means you can address the buffer directly and insert the T:PVD objects.